From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: UN1CA y2slte <un1ca-y2slte@localhost>
Date: Sat, 30 Aug 2026 00:00:00 -0300
Subject: [PATCH] Disable TWT_CONTROL support

Remove feature ID 2 (TWT) from the Android 16 default driver feature map.
---
 .../server/wifi/driver/WifiDriverFeatureProvider.smali | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/smali/com/samsung/android/server/wifi/driver/WifiDriverFeatureProvider.smali b/smali/com/samsung/android/server/wifi/driver/WifiDriverFeatureProvider.smali
--- a/smali/com/samsung/android/server/wifi/driver/WifiDriverFeatureProvider.smali
+++ b/smali/com/samsung/android/server/wifi/driver/WifiDriverFeatureProvider.smali
@@ -131,22 +131,6 @@
 .method public final configDefaultFeatureSet()V
     .locals 3

-    iget-object v0, p0, Lcom/samsung/android/server/wifi/driver/WifiDriverFeatureProvider;->mSwFeatureInteger:Ljava/util/HashMap;
-
-    const/4 v1, 0x2
-
-    invoke-static {v1}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
-
-    const/4 v2, 0x1
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v2
-
-    invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
-
     iget-object v0, p0, Lcom/samsung/android/server/wifi/driver/WifiDriverFeatureProvider;->mSwFeatureInteger:Ljava/util/HashMap;

     const/4 v1, 0x3
@@ -154,5 +138,11 @@
     invoke-static {v1}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;

     move-result-object v1

+    const/4 v2, 0x1
+
+    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
+
+    move-result-object v2
+
     invoke-virtual {v0, v1, v2}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
--
2.51.0
